experimental/packages/opentelemetry-instrumentation-http/src/utils.ts

```diff
@@ -322,6 +339,7 @@ export const getOutgoingRequestAttributes = (
     [SemanticAttributes.HTTP_METHOD]: method,
     [SemanticAttributes.HTTP_TARGET]: requestOptions.path || '/',
     [SemanticAttributes.NET_PEER_NAME]: hostname,
+    [SemanticAttributes.HTTP_HOST]: `${hostname}:${port}`,
```

Comment:
[The specification states](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#common-attributes) that this value should be filled according to the HOST header:
> The value of the [HTTP host header](https://tools.ietf.org/html/rfc7230#section-5.4). An empty Host header should also be reported, see note. [2]

It also requires that this attribute should not be set if the header is missing.
> [2]: When the header is present but empty the attribute SHOULD be set to the empty string. Note that this is a valid situation that is expected in certain cases, according the aforementioned [section of RFC 7230](https://tools.ietf.org/html/rfc7230#section-5.4). When the header is not set the attribute MUST NOT be set.

I think the current PR improves the situation, as the previous value was also not set based on headers. This fix, however, is not yet spec-compliant as the value can be set to something else than the header:
- by setting the `setHost` option to `false` on the `options` object, node's http client will not set HOST header, but attribute will be set, which violate spec: "When the header is not set the attribute MUST NOT be set."
- if port is not stated on request, node **will not** include the default port on the host header, but instrumentation **will** record the port into the attribute, which again means that the attribute is not the same as the header unlike spec.
- There is [some other logic](https://github.com/nodejs/node/blob/7cbcc4fc43ec167ae92080be5a4983821ba7ddc7/lib/_http_client.js#L276) that node uses to construct the host header which is not reflected in the current `http.host` attribute. For example: ipV6 formatting and when to iinclude port.

I wonder if is possible to use the header value directly to fill this attribute, instead of rebuilding the string (possibly, incorrectly)

Reply:
sorry for the late reply, have been busy on something else.

do you mean, we can just set the `http.host` attribute based on the `host` header? and we are ok if the value is missing or incorrect?

Reply:
I believe we can fall back to the manually formatted one if the "host" header is missing.

Reply:
Any update on this?

Reply:
updated